stories/components/form/ArrayList.stories.mdx

```diff
@@ -0,0 +1,68 @@
+import { Meta, Canvas, Story, ArgsTable, Source } from '@storybook/addon-docs';
+import ArrayList from '@/pkg/rancher-components/src/components/Form/ArrayList/ArrayList';
+import { _EDIT, _VIEW } from "@shell/config/query-params";
+
+export const Template = (args, { argTypes }) => ({
+  components: { ArrayList },
+  props:      Object.keys(argTypes),
+  template:   '<ArrayList v-bind="$props" />',
+})
+
+<Meta 
```

Comment:
We should show controls for each variation defined as default, plus your value is not set properly.

Replace this as follow:
```jsx
<Meta 
  title="Components/Form/ArrayList"
  decorators={[]}
  argTypes={{ 
    value: [],
    mode: {
      options: [_EDIT, _VIEW],
      control: { type: 'radio' },
    },
    valueLabel: 'Value',
    valuePlaceholder: 'e.g. bar',
    valueMultiline: false,
    addLabel: 'Add value',
    addAllowed: true,
    removeLabel: 'Remove value',
    removeAllowed: true,
    loading: false,
    disabled: false,
    defaultAddValue: '',
    initialEmptyRow: false,
    title: '',
    showHeader: false
  }}
/>
```

Reply:
I don't like the idea of adding all these options because it doesn't demonstrate how the component is supposed to be used. A normal use case doesn't require most of these options.

Reply:
The scope of stoybook is to interact with the component and test it there, as you do not have another places for components to do so, so any possible option should be there.
The API list is for documentation.
If you think that listing all the props may be unclear, we could add a default component for this scope.
What do you think?

Reply:
I thought tests are done in the component library and the storybook is for documentation only. I believe that documentation should be short and succinct, not overwhelming or exhaustive.

Reply:
I did not mention unit test at all. I said in Storybook you have to see how the component works in its integrity like in any styleguide of any UI library existing. That's literally the only way to know how to use them.
https://vuetifyjs.com/en/components/

Reply:
Beside the fact, visual tests in Storybook (as it comes out of the box) should also be something in an optimal scenario.
Some people do also screenshot testing through Jest, although it requires more efforts.